### PR TITLE
Allow getting `AccelOutputDataRate` and `MagOutputDataRate` from Hertz.

### PR DESCRIPTION
**Pull request by reitermarkus**

### CODE REVIEW

File: src/types.rs [hunk 1]
```diff
@@ -214,29 +214,46 @@ impl MagneticField {
...
+    /// 1.620 kHz (only low-power mode)
     Khz1_620LowPower,
-    /// 5.376 kHz (Low-power)
+    /// 5.376 kHz (only low-power mode)
     Khz5_376LowPower,
 }
 
 impl AccelOutputDataRate {
+    /// Create an `AccelOutputDataRate` with the given frequency in Hertz.
+    pub const fn from_freq(hz: u16) -> Option<Self> {
```
Comment (eldruin) [inline src/types.rs:241]: How about implementing `TryFrom` instead? sounds more idiomatic.

**Review by eldruin — COMMENTED** [review]
Thanks!

File: src/types.rs (hunk 1 above)
Comment (reitermarkus) [reply src/types.rs:241]: `TryFrom<u16>` is a bit ambiguous in what exactly the `u16` represents. Also, `TryFrom` is not `const` currently. Arguably, this should probably be called `from_hertz`.

Comment (eldruin) [reply src/types.rs:241]: I understand. Could you rename the method `from_hertz`, then? It does sound better.
Also, there is a conflict now that I merged #15.

**Review by eldruin — COMMENTED** [review]
Since the tests only use the public interface, could you move them into a file in the tests folder?
Also, could you add some more (possibly via a macro to avoid so much repetition) for the other values?
The coverage keeps [sliding down](https://coveralls.io/builds/51552978/source?filename=src%2Ftypes.rs#L279)

**Review by eldruin — APPROVED** [review]
Looks good to me, thanks!

**Pull request merged**